packages/gensx-claude-md/templates/CLAUDE.md

```diff
@@ -36,84 +34,123 @@ npm run start
 
 ### Component Definition
 
+Components are the building blocks of GenSX applications. Here's a basic example of a GenSX component:
+
 ```typescript
-interface GreetingProps {
+import * as gensx from "@gensx/core";
+import { openai } from "@ai-sdk/openai";
+import { generateText } from "@gensx/vercel-ai";
+
+interface GreetingInput {
   name: string;
-  formatOutput?: boolean;
+  formal?: boolean;
 }
 
-const Greeting = gensx.Component<GreetingProps, string>(
+const Greeting = gensx.Component(
   "Greeting",
-  async ({ name, formatOutput = false }) => {
-    const greeting = `Hello, ${name}!`;
-    return formatOutput ? `## ${greeting}` : greeting;
+  async ({ name, formatOutput = false, formal = false }: GreetingInput) => {
+    const result = await generateText({
+      model: openai("gpt-4.1-mini"),
+      messages: [
+        {
+          role: "system",
+          content:
+            "You are a friendly assistant that creates personalized greetings. You can adjust your tone based on whether the greeting should be formal or informal.",
+        },
+        {
+          role: "user",
+          content: `Create a ${formal ? "formal" : "casual"} greeting for ${name}. Keep it concise and ${formal ? "professional" : "warm"}.`,
+        },
+      ],
+    });
+    return result.text;
   },
 );
 ```
 
-### Provider Usage
+To run a component, you just call it like a function:
 
 ```typescript
-const MyWorkflow = gensx.Component(
-  "MyWorkflow",
-  () => (
-    <OpenAIProvider apiKey={process.env.OPENAI_API_KEY}>
-      <ChatCompletion
-        model="gpt-4o-mini"
-        messages={[
-          { role: "system", content: "You are a helpful assistant." },
-          { role: "user", content: "Tell me about GenSX." }
-        ]}
-      />
-    </OpenAIProvider>
-  )
-);
+const greeting = await Greeting({ name: "John", formal: true });
 ```
 
-### Component Nesting with Child Functions
+### Workflow Definition
+
+In GenSX, workflows are the entry points to your application. Here's a basic example of a GenSX workflow:
```

Comment:
I don't love this verbiage but couldn't really think of anything better at the moment; would love suggestions.

Reply:
Here is a suggestion on how to differentiate the two: 

- Components are reusable workflow steps. They are a unit of tracing, and the inputs and outputs of the component will be recorded and traced
- Workflows are collections of components. Each workflow will create a top level trace that shows all components that executed within it. Workflows are automatically converted into API endpoints for synchronous and asyncronous invocation when exported from the file (show example).